src/components/inputs/checkbox-input/checkbox-input.js

```diff
@@ -63,41 +98,36 @@ class CheckboxInput extends React.PureComponent {
   });
 
   render() {
+    const disabled = this.props.disabled || this.props.isDisabled;
+
     return (
-      <Label htmlFor={this.state.id} hasError={this.props.hasError}>
-        <Checkbox
-          type="checkbox"
-          id={this.state.id}
-          name={this.props.name}
-          value={this.props.value}
-          onChange={this.props.onChange}
-          isDisabled={this.props.isDisabled}
-          isChecked={this.props.isChecked}
-          isIndeterminate={this.props.isIndeterminate}
-          {...filterDataAttributes(this.props)}
-          {...filterAriaAttributes(this.props)}
-        />
-        <div css={theme => getCheckboxWrapperStyles(this.props, theme)}>
-          {(() => {
-            if (this.props.isIndeterminate) return <Icons.Indeterminate />;
-            if (this.props.isChecked) return <Icons.Checked />;
-            return <Icons.Unchecked />;
-          })()}
-        </div>
-        {this.props.children && (
-          <div
-            css={css`
-              margin-left: ${vars.spacingS};
-            `}
-          >
+      <Label
+        htmlFor={this.state.id}
+        disabled={disabled}
+        hasError={this.props.hasError}
+      >
+        <Checkbox id={this.state.id} {...getCheckboxProps(this.props)} />
+        <Spacings.Inline alignItems="center" scale="s">
```

Comment:
Removed the styled `div` in L88 and added this wrapper here for the same effect.